src/index.d.ts

```diff
@@ -14,6 +14,11 @@ interface ReactI18nextModule {
   init: (instance: i18next.i18n) => void;
 }
 
+interface ReactI18nextTranslateFunction {
+  t<TKeys extends string=string,TVals extends object =object,TResult=any>
```

Comment:
It was migrated to i18next https://github.com/i18next/i18next/blob/master/index.d.ts#L463-L480 . 
Default value is set any before I imprement tha,but I understood TrResult may be more strict as you mean.

Reply:
I agree default value was set any before your implementation. But since you're working on the generic, it was perfect time to follow the boyscout rule. There is so much any in these typings.

Reply:
@VincentLanglet I take a different tact.  Incremental improvement.  `any` is something and if you feel strongly about improving the type then PR it.  Something is better than nothing, in this case `any` is better than no type at all so I don't pile on and ask for more changes that are near but not necessarily the intent.  Otherwise it would spider into fixing all `any` types.  I encourage small focused PRs if possible.  Review/test/merge them quickly and let everyone contribute to their concern.

Reply:
I'm not talking about fixing all any types. Just the one he's introducing.
You ask for my contribution, but this code isn't already in the base code.
When we write code, we write test too, we don't explain "Someone can do the test for me". It's the same here.

I don't file it's difficult to change TResult = any by something like TResult extends string | string[] = string | string[].

Reply:
Thanks @VincentLanglet - you just proved my point and incrementally improved the project!